components/script/body.rs

```diff
@@ -27,14 +32,16 @@ pub enum BodyType {
     Blob,
     FormData,
     Json,
-    Text
+    Text,
+    ArrayBuffer
 }
 
 pub enum FetchedData {
     Text(String),
-    Json(JSValue),
+    Json(RootedTraceableBox<Heap<JSValue>>),
```

Comment:
Updated Json to store `RootedTraceableBox<Heap>` as suggested. Still it stores `JSValue` instead of `JSVal`, by `JSVal` doesn't meet trait bounds for  `RootedTraceableBox`.